jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -4588,15 +4588,19 @@ public boolean isNestmateOf(Class<?> that) {
 	if (nestMembers == null) {
 		nestMembers = getNestMembersImpl();
 	}
-	SecurityManager securityManager = System.getSecurityManager();
-	if (securityManager != null) {
-		/* All classes in a nest must be in the same runtime package and therefore same classloader */
-		ClassLoader nestMemberClassLoader = this.internalGetClassLoader();
-		ClassLoader callerClassLoader = ClassLoader.getCallerClassLoader();
-		if (!doesClassLoaderDescendFrom(nestMemberClassLoader, callerClassLoader)) {
-			String nestMemberPackageName = this.getPackageName();
-			if ((nestMemberPackageName != null) && (nestMemberPackageName != "")) {
-				securityManager.checkPackageAccess(nestMemberPackageName);
+	if (nestMembers.length == 1 && nestMembers[0] == this) {
+		/* this is a Class object representing array types, primitive types, or void */
```

Comment:
Or a class in its own nest.

Reply:
Right, updated the comment, please have another look.